client/input.go

```diff
@@ -14,10 +14,26 @@ import (
 // MinPassLength is the minimum acceptable password length
 const MinPassLength = 8
 
+var currentStdin *bufio.Reader
```

Comment:
IMO this belongs to `CLIContext`

Reply:
The problem with context is that moving things there results in cyclic dependencies with `keys`. I will need further refactoring to make `context` independant from other packages.
 
The original location of `func BufferStdin() *bufio.Reader` and in order to inject a different stdin I placed it as near as possible.

I would suggest we create another PR to make context a first-class citizen :) Including stdout too.

Reply:
I've just created an issue for this: 
https://github.com/cosmos/cosmos-sdk/issues/3558

Reply:
> Including stdout too.

And stderr :)